packages/core-js/modules/esnext.promise.all-settled.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+// `Promise.allSettled` method
+// https://github.com/jasonwilliams/proposal-promise-allSettled
+var newPromiseCapabilityModule = require('../internals/new-promise-capability');
+var perform = require('../internals/perform');
+var iterate = require('../internals/iterate');
+
+require('../internals/export')({ target: 'Promise', stat: true }, {
+  allSettled: function allSettled(iterable) {
+    var C = this;
+    var capability = newPromiseCapabilityModule.f(C);
+    var resolve = capability.resolve;
+    var reject = capability.reject;
+    var result = perform(function () {
+      var values = [];
+      var counter = 0;
+      var remaining = 1;
+      iterate(iterable, function (promise) {
+        var index = counter++;
+        var alreadyCalled = false;
+        values.push(undefined);
+        remaining++;
+        C.resolve(promise).then(function (value) {
+          if (alreadyCalled) return;
+          alreadyCalled = true;
+          values[index] = { value: value, status: 'fulfilled' };
+          --remaining || resolve(values);
+        }, function (e) {
+          if (alreadyCalled) return;
+          alreadyCalled = true;
+          values[index] = { reason: e, status: 'rejected' };
+          --remaining || resolve(values);
+        });
+      });
+      --remaining || resolve(values);
+    });
+    if (result.e) reject(result.v);
+    return capability.promise;
+  }
+});
```

Comment:
cc @jasonwilliams @mathiasbynens internal logic is similar to `Promise.all`. Any comments?

Reply:
@zloirock looks good to me, i don't see any issues there.